syntax = "proto3";

package datacloud.network.v1;

import "datacloud/v1/operation.proto";
import "datacloud/v1/paging.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/datacloud/private_api/datacloud/network/v1;network";
option java_package = "doublecloud.priv.network.v1";
option java_outer_classname = "DNOS";

// A set of methods for managing operations that are asynchronous API requests.
service OperationService {
    // Returns the specified operation.
    rpc Get (GetOperationRequest) returns (datacloud.v1.Operation);

    // Retrieves a list of operations that belong to the specified project.
    rpc List (ListOperationsRequest) returns (ListOperationsResponse);
}

message GetOperationRequest {
    // Required. ID of the operation to return.
    string operation_id = 1;
}

message ListOperationsRequest {
    // ID of the project to list clusters operations in.
    string project_id = 1;

    // Paging information of the request
    datacloud.v1.Paging paging = 2;
}

message ListOperationsResponse {
    // List of operations.
    repeated datacloud.v1.Operation operations = 1;

    // Pagination information of the response
    datacloud.v1.NextPage next_page = 2;
}
